src/qdrant/qdrant_datasink.py

```diff
@@ -1,71 +1,78 @@
-import logging
-from typing import TYPE_CHECKING, Dict, List, Any, Iterable, Optional
+from typing import Any, Iterable, Optional
 
 from ray.data.block import Block
 from ray.data.datasource.datasink import Datasink
 from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
 from ray.data._internal.execution.interfaces import TaskContext
-from ray.data._internal.util import _check_import
 from ray.data.block import BlockAccessor
-from ray.util.annotations import PublicAPI
 
 from qdrant_client import QdrantClient
 from qdrant_client.models import PointStruct
 
-logger = logging.getLogger(__name__)
 
 class QdrantDatasink(Datasink):
-    def __init__(self, host: str, port: int, collection_name: str, batch_size: int = 100) -> None:
-        self.host = host
-        self.port = port
-        self.collection_name = collection_name
-        self.batch_size = batch_size
-        self._client = None
+    def __init__(
+        self,
+        client: QdrantClient,
+        collection_name: str,
+        batch_size: int = 64,
+        parallel: int = 1,
+        **kwargs: Any,
+    ) -> None:
+        self._collection_name = collection_name
+        self._batch_size = batch_size
+        self._parallel = parallel
+        self._kwargs = kwargs
+
+        # Can't set QdrantClient directly due to serialization issues
+        self._client_opts = client.init_options
```

Comment:
`init_options` property will be available from the next release of `qdrant-client`. Will have to wait.

Reply:
Out now.

Reply:
Great I’ll try it out tonight and merge it if I don’t see any issues. Thanks.